pkg/graveler/retention/expired_commits.go

```diff
@@ -12,37 +14,68 @@ type GarbageCollectionCommits struct {
 	active  []graveler.CommitID
 }
 
+type CommitNode struct {
+	CommitID     graveler.CommitID
+	CreationDate time.Time
+	ParentsIDs   []graveler.CommitID
+	Version      graveler.CommitVersion
```

Comment:
```suggestion
	CreationDate time.Time
	MainParent   graveler.CommitID
```
1. Version is the key of the map.
2. Find the main parent in the pre-processing.